internal/services/workloads/workloads_sap_three_tier_virtual_instance_resource.go

```diff
@@ -0,0 +1,1943 @@
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+
+								"database_server": {
+									Type:     pluginsdk.TypeList,
+									Optional: true,
+									ForceNew: true,
+									MaxItems: 1,
+									Elem: &pluginsdk.Resource{
+										Schema: map[string]*pluginsdk.Schema{
+											"availability_set_name": {
+												Type:         pluginsdk.TypeString,
+												Optional:     true,
+												ForceNew:     true,
+												ValidateFunc: validation.StringLenBetween(1, 80),
+											},
+
+											"load_balancer": {
+												Type:     pluginsdk.TypeList,
+												Optional: true,
+												ForceNew: true,
+												MaxItems: 1,
+												Elem: &pluginsdk.Resource{
+													Schema: map[string]*pluginsdk.Schema{
+														"name": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: validation.StringIsNotEmpty,
+														},
+
+														"backend_pool_names": {
+															Type:     pluginsdk.TypeList,
+															Optional: true,
+															ForceNew: true,
+															Elem: &pluginsdk.Schema{
+																Type:         pluginsdk.TypeString,
+																ValidateFunc: validation.StringIsNotEmpty,
+															},
+														},
+
+														"frontend_ip_configuration_names": {
+															Type:     pluginsdk.TypeList,
+															Optional: true,
+															ForceNew: true,
+															Elem: &pluginsdk.Schema{
+																Type:         pluginsdk.TypeString,
+																ValidateFunc: validation.StringIsNotEmpty,
+															},
+														},
+
+														"health_probe_names": {
+															Type:     pluginsdk.TypeList,
+															Optional: true,
+															ForceNew: true,
+															Elem: &pluginsdk.Schema{
+																Type:         pluginsdk.TypeString,
+																ValidateFunc: validation.StringIsNotEmpty,
+															},
+														},
+													},
+												},
+											},
+
+											"virtual_machine": {
+												Type:     pluginsdk.TypeList,
+												Optional: true,
+												ForceNew: true,
+												Elem: &pluginsdk.Resource{
+													Schema: map[string]*pluginsdk.Schema{
+														"data_disk": {
+															Type:     pluginsdk.TypeSet,
+															Optional: true,
+															ForceNew: true,
+															Elem: &pluginsdk.Resource{
+																Schema: map[string]*pluginsdk.Schema{
+																	"volume_name": {
+																		Type:     pluginsdk.TypeString,
+																		Required: true,
+																		ForceNew: true,
+																		ValidateFunc: validation.StringInSlice([]string{
+																			"hanaData",
+																			"hanaLog",
+																			"hanaShared",
+																			"usrSap",
```

Comment:
are these validation options applicable to the `data_disk` properties on lines 633 and 765 too?

Reply:
I can confirm that these validation options aren't applicable to the `data_disk` properties on lines 633 and 765.